4_gap_analysis/src/Con_byState.R

```diff
@@ -115,29 +123,34 @@ sample_rates <- samples_per_year %>%
     MeanSamplesPerYear=mean(NumSamplesPerYear),
     MinSamplesPerYear=min(NumSamplesPerYear),
     MaxSamplesPerYear=max(NumSamplesPerYear),
-    MedianSamplesPerYear=median(NumSamplesPerYear))
+    MedianSamplesPerYear=median(NumSamplesPerYear),
+    NoYears=n_distinct(Year))
 
 # split into scenarios
 # Do some more work here
-scenario_empty <- sample_rates %>%
-  filter(MinNumConstituents == 3, MinSamplesPerYear >= 12)
 scenario_A <- sample_rates %>%
-  filter(MinNumConstituents == 2, MinSamplesPerYear >= 12)
+  filter(MinNumConstituents == 3, MinSamplesPerYear >= 12, NoYears>4)
 scenario_B <- sample_rates %>%
-  filter(MinNumConstituents == 1, MinSamplesPerYear >= 12)
+  filter(MinNumConstituents < 3, MinSamplesPerYear >= 12, NoYears>4)
 scenario_C <- sample_rates %>%
-  filter(MinNumConstituents == 2, MinSamplesPerYear >= 6)
+  filter(MinNumConstituents == 3, MinSamplesPerYear >= 6 & MinSamplesPerYear < 12, NoYears>4)
+scenario_D <- sample_rates %>%
+  filter(MinNumConstituents < 3, MinSamplesPerYear >= 6 & MinSamplesPerYear < 12, NoYears>4)
+scenario_E <- sample_rates %>%
+  filter(MinNumConstituents == 3, MinSamplesPerYear >= 4 & MinSamplesPerYear < 6, NoYears>4)
+
+siteInfo<-rbind(scenario_A,scenario_B,scenario_C,scenario_D,scenario_E)
+siteInfo$cat<-ifelse(siteInfo$MonitoringLocationIdentifier %in% scenario_A$MonitoringLocationIdentifier,'A) 3 constituents (con.), >= 12 samples/year',
```

Comment:
this looks like it works fine, but it's a lot of text - you could probably make it more concise with `dplyr::case_when` (https://dplyr.tidyverse.org/reference/case_when.html)